docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_s3_bucket_remediation.mdx

```diff
@@ -1,6 +1,203 @@
+
+2. **Create or Modify a Trail:**
+   - If you already have a trail, select it from the list. If not, click on "Create trail" to set up a new one.
+
+3. **Configure Management Events:**
+   - In the trail settings, locate the "Management events" section.
+   - Ensure that the "Read/Write management events" option is enabled. You can choose to log "Read events," "Write events," or both, depending on your requirements.
+
+4. **Save Changes:**
+   - After configuring the management events, review your settings.
+   - Click on "Create" or "Update trail" to save the changes.
+
+By following these steps, you ensure that CloudTrail is configured to log management events, which is crucial for auditing and monitoring activities in your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that CloudTrail logs management events in AWS, you need to configure your CloudTrail settings appropriately. Here are the steps to prevent the misconfiguration using AWS CLI:
+
+1. **Create or Update a CloudTrail Trail:**
+   Ensure you have a CloudTrail trail that is configured to log management events. If you don't have one, create it using the following command:
+   ```sh
+   aws cloudtrail create-trail --name <trail-name> --s3-bucket-name <s3-bucket-name>
+   ```
+
+2. **Enable Management Event Logging:**
+   When creating or updating the trail, make sure to enable management event logging. Use the following command to update an existing trail to log management events:
+   ```sh
+   aws cloudtrail update-trail --name <trail-name> --is-logging --include-management-events
+   ```
+
+3. **Verify Management Event Logging Configuration:**
+   Check the trail configuration to ensure that management events are being logged. Use the following command to describe the trail and verify the settings:
+   ```sh
+   aws cloudtrail describe-trails --trail-name-list <trail-name>
+   ```
+
+4. **Enable Multi-Region Trail (Optional but Recommended):**
+   To ensure that management events are logged across all regions, enable the multi-region trail option:
+   ```sh
+   aws cloudtrail update-trail --name <trail-name> --is-multi-region-trail
+   ```
+
+By following these steps, you can ensure that CloudTrail is configured to log management events, thereby preventing the misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that CloudTrail logs management events in AWS, you can use the Boto3 library in Python. Here are the steps to prevent the misconfiguration where CloudTrail does not log management events:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with the necessary AWS credentials and region.
+
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>